x/xio/types.go

```diff
@@ -22,16 +22,27 @@ package xio
 
 import (
 	"io"
+	"time"
 
-	"github.com/m3db/m3x/resource"
 	"github.com/m3db/m3db/ts"
+	"github.com/m3db/m3x/resource"
 )
 
-// ReaderSliceReader implements the io reader interface backed by a slice of readers
-type ReaderSliceReader interface {
-	io.Reader
+// BlockReader implements the io reader interface backed by a segment with start and end times
+type BlockReader interface {
```

Comment:
was talking to Rob offline: could you change this so BlockReader is just a struct?

```
type BlockReader struct {
  Start time.Time
  End time.Time
  SegmentReader SegmentReader
}
```